### PR TITLE
Removes unnecessary integration tests

### PR DESCRIPTION
**Pull request by ForestEckhardt**
- Removes unused fixtures
- Removes tests who functionality I feel is sufficiently covered by existing unit testing or tested in other buildpacks

## Checklist
* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have added an integration test, if necessary.

### CODE REVIEW

**Comment by sophiewigmore** [comment]
Since there's quite a few changed files, I went through and made a summary of what you got rid of in this PR:

Integration tests removed:
-  integration/aspnet_test.go
- integration/kestrel_app_test.go
- integration/versioning_test.go

fixtures removed:

- with_dot_in_name
- uses_libgdiplus - (makes sense since we don't support libgdi plus)
- source_prerender_node

- source_aspnetcore_app_2.1  (used in deleted test)
- source_aspnetcore_all_2.1 (used in deleted test)
- source_2.1_explicit_runtime (used in deleted test)
- source_2.1_float_runtime (used in deleted test)
- nancy_kestrel_msbuild_dotnet2 (used in deleted test)

**Comment by ForestEckhardt** [comment]
@sophiewigmore 
* The kestrel apps build. They are just only configurable through `ASPNETCORE_URLS` I am not really concerned with how the users need to configure the apps as long as we build them functionally.
* The `with_dot_in_name` fixture had no associated tests and I could just build the app and it functioned. I see no reason why the app was majorly different. I think that it should be removed because the worst case scenario is a bug report somewhere down the road.
* The `source_prerendered_node` fixture is one that I was able to build independently using the `build-plan` buildpack. I would like to hold off on testing this until we know that it is a somewhat common use case that should be supported natively by the buildpack.

**Comment by sophiewigmore** [comment]
> * The kestrel apps build. They are just only configurable through `ASPNETCORE_URLS` I am not really concerned with how the users need to configure the apps as long as we build them functionally.

Maybe the integration tests aren't the right place for this, but I do think it would be helpful if we had an example somewhere that showed an example of a kestrel app working, and that ASPNETCORE_URLS` is needed.

> * The `with_dot_in_name` fixture had no associated tests and I could just build the app and it functioned. I see no reason why the app was majorly different. I think that it should be removed because the worst case scenario is a bug report somewhere down the road.

All good with this reasoning!

> * The `source_prerendered_node` fixture is one that I was able to build independently using the `build-plan` buildpack. I would like to hold off on testing this until we know that it is a somewhat common use case that should be supported natively by the buildpack.

For this I'm on the fence, since this case is something that the V2Bs covers and the Node Engine buildpack is in the language family order groupings but we don't have any tests besides this one (that I'm aware of?) that test a .NET Core app with Node. I feel like if it's an order grouping we support, we should test it _somewhere_

**Comment by fg-j** [comment]
@ForestEckhardt I think that "with dot in name" fixture might have had something to do with TAS4K8s having issues building apps with dots in the name.

@sophiewigmore 

> For this I'm on the fence, since this case is something that the V2Bs covers and the Node Engine buildpack is in the language family order groupings but we don't have any tests besides this one (that I'm aware of?) that test a .NET Core app with Node. I feel like if it's an order grouping we support, we should test it somewhere

As far as I know, we test node in the order group in the [language family integration tests](https://github.com/paketo-buildpacks/dotnet-core/tree/main/integration/testdata/source-app) which build a React app from source. Is the node fixture here different in some way from that type of node app?

File: integration/testdata/source_2.1_explicit_runtime/netcoreapp2.csproj [hunk 1]
```diff
@@ -1,16 +0,0 @@
-<Project Sdk="Microsoft.NET.Sdk.Web">
-
-  <PropertyGroup>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
-    <RuntimeFrameworkVersion>2.1.0</RuntimeFrameworkVersion>
```
Comment (fg-j) [inline integration/testdata/source_2.1_explicit_runtime/netcoreapp2.csproj:5]: Seems like the crux of the versioning test is the difference between this (explicit) version and the "floating" runtime version specified [in the other fixture's csproj](https://github.com/paketo-buildpacks/dotnet-publish/blob/c49c431209299a84536713b40a2ac1bb0b6d8708/integration/testdata/source_2.1_float_runtime/netcoreapp2.csproj#L5). 

This difference seems related to [this issue](https://github.com/paketo-buildpacks/dotnet-core-runtime/issues/135) about roll-forward logic. But the tests here don't make any assertions about the runtime version installed. [We have tests in runtime](https://github.com/paketo-buildpacks/dotnet-core-runtime/blob/cec9c25e5ce73128d3f600809610a90f9c475959/runtime_version_resolver_test.go#L183) that figure out the proper version to install given constraints from the projfile. I agree, don't think we need to test this here.

File: integration/aspnet_test.go [hunk 2]
```diff
@@ -1,129 +0,0 @@
...
-		})
-
-		it.After(func() {
-			Expect(docker.Container.Remove.Execute(container.ID)).To(Succeed())
-			Expect(docker.Image.Remove.Execute(image.ID)).To(Succeed())
-			Expect(docker.Volume.Remove.Execute(occam.CacheVolumeNames(name))).To(Succeed())
-			Expect(os.RemoveAll(source)).To(Succeed())
-		})
-
-		it("should build a working OCI image for a source_aspnetcore_all_2.1 application", func() {
```
Comment (fg-j) [inline integration/aspnet_test.go:52]: Seems like these tests are supposed to make sure that the dotnet-publish buildpack correctly `requires` ASPNET when it finds certain package references ([AspNetCore.App](https://github.com/paketo-buildpacks/dotnet-publish/blob/c49c431209299a84536713b40a2ac1bb0b6d8708/integration/testdata/source_aspnetcore_app_2.1/source_aspnetcore_2.1.csproj#L12) and [AspNetCore.All](https://github.com/paketo-buildpacks/dotnet-publish/blob/c49c431209299a84536713b40a2ac1bb0b6d8708/integration/testdata/source_aspnetcore_all_2.1/netcoreapp2.csproj#L8)) in the projfile. However this detection logic is [covered in the project file parser tests](https://github.com/paketo-buildpacks/dotnet-publish/blob/main/dotnet_project_file_parser_test.go#L117-L152) so I don't think we need an integration test.

**Comment by fg-j** [comment]
As for the kestrel test, looks like [it was ported over directly from the v2 fixtures](https://github.com/paketo-buildpacks/dotnet-publish/commit/d49c78b46cfe2f882c53a929dd109126aeddca87). As long as we're confident there is no special detection criteria or build differences, I think it makes sense to get rid of.

**Review by fg-j — APPROVED** [review]

**Review by sophiewigmore — APPROVED** [review]

**Pull request merged**